fdbserver/workloads/SnapTest.actor.cpp

```diff
@@ -96,7 +96,10 @@ struct SnapTestWorkload : TestWorkload {
 
 	void getMetrics(std::vector<PerfMetric>& m) override { TraceEvent("SnapTestWorkloadGetMetrics"); }
 
-	void disableFailureInjectionWorkloads(std::set<std::string>& out) const override { out.insert("all"); }
+	void disableFailureInjectionWorkloads(std::set<std::string>& out) const override {
+		// Data movement is not allowed while taking snapshot
+		out.insert("RandomMoveKeys");
+	}
```

Comment:
Does running `RandomMoveKeys` while taking a snapshot cause the test to fail? In theory, the data movement should just get delayed.

Reply:
Yeah, it's causing a very complicated failure where the data is moved while taking snapshots, and in the second phase, a moved range is not found, and test gets stuck.

Reply:
Can you add a `FIXME` comment here to reenable this fault injection workload in the future? For now, it shouldn't be a blocker for merging this PR, since this PR strictly improves our test coverage for taking snapshots.

Reply:
Sure, I added the comment.
I think the issue after discussion with @sfc-gh-xwang is that the `RandomMoveKeys` doesn't know that the snapshot disables DD.
In particular, the [DDEnabledState](https://github.com/apple/foundationdb/blob/63354f68ad78e54854c8dc4217c5e01b729667a6/fdbserver/workloads/RandomMoveKeys.actor.cpp#L158) used by the workload is just a local state, which is not the same one we use on DD